pkg/server/server.go

```diff
@@ -0,0 +1,191 @@
+package server
+
+import (
+	configmodels "abelgoodwin1988/iata-finder/configs/models"
+	"abelgoodwin1988/iata-finder/pkg/dataservice"
+	"abelgoodwin1988/iata-finder/pkg/logger"
+	iatafinder "abelgoodwin1988/iata-finder/rpc"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+)
+
+var ctxLogger = logger.CtxLogger
+var ds *dataservice.Dataservice
+
+type server struct{}
+
+// Create initializes the rpc server
+func Create(dataservice *dataservice.Dataservice, configPath string) (net.Listener, *grpc.Server) {
+	ctxLogger.Info("Starting iata-finder service")
+
+	if dataservice == nil {
+		// Will warn for now
+		ctxLogger.Warningln("No Dataservice provided")
+	}
+
+	ds = dataservice
+
+	// load config values for rpc
+	rpcConfig := configmodels.RPCConfig{}
+	loadRPCConfig(&rpcConfig, configPath)
+
+	// Configure and create rpc server
+	lis := rpcListen(&rpcConfig)
+
+	s := grpc.NewServer()
+	iatafinder.RegisterIatafinderServer(s, &server{})
+
+	ctxLogger.WithFields(logrus.Fields{"at": fmt.Sprintf("%s:%s", rpcConfig.IP, rpcConfig.Port)}).Info("iata-finder service configured to be served")
+
+	// return listener and registered server. We will serve outside of server
+	// since we may need to run server inside a go routine in test case
+	return lis, s
+}
+
+func loadRPCConfig(rpcConfig *configmodels.RPCConfig, configPath string) {
+	data, err := ioutil.ReadFile(configPath)
+	if err != nil {
+		ctxLogger.WithError(err).Error("Failed to read rpc.config.go")
+	}
+	if _, err := toml.Decode(string(data), &rpcConfig); err != nil {
+		ctxLogger.WithError(err).Error("Failed to decode rpc.config.go")
+	}
+}
+
+func rpcListen(rpcConfig *configmodels.RPCConfig) net.Listener {
+	network := rpcConfig.Network
+	ip := rpcConfig.IP
+	port := rpcConfig.Port
+	address := fmt.Sprintf("%s:%s", ip, port)
+	lis, err := net.Listen(network, address)
```

Comment:
Same thing here; we don't need to redeclare all this when it's already available.

I think in -most- languages we like to do this because we're used to loading up the garbage collector and the language necessitates clarity and simplification through doing stuff like this...

Again, it doesn't really matter but I dont think it's idiomatic. This is probably something I wrote, huh? 😂